src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,307 @@
+        if (Object.keys(argMap).indexOf(orgName) > -1) {
+            return argMap[orgName];
+        } else {
+            return orgName;
+        }
+    };
+    /**********************The functions below Add Connections between the Layers**************/
+    ImportKeras.prototype.addConnections = function () {
+        // this._findNodeByName();
+        let layers = this.modelInfo.layers;
+        let layerInputConnections = {};
+        let layerOutputConnections = {};
+        let connections = null;
+        layers.forEach((layer) => {
+            layerInputConnections[layer.name] = [];
+            layerOutputConnections[layer.name] = [];
+        });
+
+        layers.forEach((layer) => {
+            if (layer.inbound_nodes.length > 0) {
+                connections = layer.inbound_nodes;
+                connections.forEach((connection) => {
+
+                        if (this._layerNameExists(connection)) {
+                            layerInputConnections[layer.name].push(connection);
+                            layerOutputConnections[connection].push(layer.name);
+                    }
+                });
+
+            }
+        });
+
+        return this._updateConnections(layerInputConnections, layerOutputConnections);
+
+    };
+
+
+    ImportKeras.prototype._layerNameExists = function (layerName) {
+        let allLayerNames = this.modelInfo.layers.map((layer) => {
+            return layer.name;
+        });
+
+        return allLayerNames.indexOf(layerName) > -1;
+    };
+
+    ImportKeras.prototype._updateConnections = function (inputs, outputs) {
+        let allLayerNames = Object.keys(inputs);
+        return Promise.all(allLayerNames.map((layerName) => {
+            let dstLayer = this.layerInfo[layerName];
+            let srcs = inputs[layerName];
+            if (srcs.length > 0) {
+                return Promise.all(srcs.map((src, index, records) => {
+                    return this._connectLayers(this.layerInfo[src], dstLayer, index);
+                }));
+            }
+
+        }));
+    };
+
+    ImportKeras.prototype._connectLayers = async function (srcLayer, dstLayer, index) {
+
+        let srcPort = await this.core.loadMembers(srcLayer, 'outputs');
+        let dstPort = await this.core.loadMembers(dstLayer, 'inputs');
+        if (dstPort && srcPort) {
+            this.core.addMember(dstPort[0], 'source', srcPort[0]);
+            this.core.setMemberRegistry(dstPort[0],
+                'source',
+                this.core.getPath(srcPort[0]),
+                'position', {x: 100, y: 100});
+            this.core.setMemberAttribute(dstPort[0], 'source',
+                this.core.getPath(srcPort[0]),
+                'index', index);
+            this.logger.debug(`Connected ${this.core.getAttribute(srcLayer, 'name')} ` +
+                `with ${this.core.getAttribute(dstLayer, 'name')} as input ${index}`);
+            return Promise.resolve('Done');

```

Comment:
Don't make a synchronous function async w/o good reason.

Reply:
`_connectLayers` should be async in my view, because `this.core.loadMembers()` returns a `Promise`. But, since `async` functions already return a promise, `return Promise.resolve('Done');` is not needed.
```javascript
    ImportKeras.prototype._connectLayers = async function (srcLayer, dstLayer, index) {

        let srcPort = await this.core.loadMembers(srcLayer, 'outputs');
        let dstPort = await this.core.loadMembers(dstLayer, 'inputs');
        if (dstPort && srcPort) {
            this.core.addMember(dstPort[0], 'source', srcPort[0]);
            this.core.setMemberRegistry(dstPort[0],
                'source',
                this.core.getPath(srcPort[0]),
                'position', {x: 100, y: 100});
            this.core.setMemberAttribute(dstPort[0], 'source',
                this.core.getPath(srcPort[0]),
                'index', index);
            this.logger.debug(`Connected ${this.core.getAttribute(srcLayer, 'name')} ` +
                `with ${this.core.getAttribute(dstLayer, 'name')} as input ${index}`);
        }
    };
```